### PR TITLE
docs(storybook): integrate compodoc for enhanced documenation

### PR DESCRIPTION
**Pull request by codexshell**
Installed Compodoc and configured its integration with Storybook. This change allows for automated documentation generation, providing a detailed and interactive documentation environment for the project's components.

This provides a consistent DX, similar to what one would find in other frameworks (libraries included).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,5 +50,6 @@
     "react-dom": "^18.2.0",
     "storybook": "^7.6.17",
     "typescript": "~5.3.2"
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha1.ac34549e6aa8e7ead463a7407e1c7390f61a6610"
```
Comment (brandonroberts) [inline package.json:54]: Can we remove this line? There's already a `package-lock.json`

**Comment by brandonroberts** [comment]
Cool, thanks for adding this @codexshell!

File: package.json (hunk 1 above)
Comment (codexshell) [reply package.json:54]: Yeah, sure. Thanks for pointing that out..

**Comment by codexshell** [comment]
Hi @brandonroberts, I've made the requested changes. Let me know if there is anything else that may need attention.

**Pull request merged**